[BUG]: variable cannot appear in its own initializer

### Version

24.10

### Which installation method(s) does this occur on?

Source

### Describe the bug.

This line of code is a non-sense:

https://github.com/rapidsai/cugraph/blob/branch-24.12/cpp/src/c_api/core_number.cpp#L79



### Minimum reproducible example

_No response_

### Relevant log output

core_number.cpp:79:75: error: variable 'degree_type' declared with deduced type 'auto' cannot appear in its own initializer
   79 |       auto degree_type = reinterpret_cast<cugraph::k_core_degree_type_t>(degree_type)

### Environment details

_No response_

### Other/Misc.

_No response_

### Code of Conduct

- [x] I agree to follow cuGraph's Code of Conduct
- [x] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/cpp/src/c_api/core_number.cpp
+++ b/cpp/src/c_api/core_number.cpp
@@ -76,7 +76,7 @@ struct core_number_functor : public cugraph::c_api::abstract_functor {
       rmm::device_uvector<edge_t> core_numbers(graph_view.local_vertex_partition_range_size(),
                                                handle_.get_stream());
 
-      auto degree_type = reinterpret_cast<cugraph::k_core_degree_type_t>(degree_type);
+      auto degree_type = reinterpret_cast<cugraph::k_core_degree_type_t>(degree_type_);
 
       cugraph::core_number<vertex_t, edge_t, multi_gpu>(handle_,
                                                         graph_view,

